apps/vmq_enhanced_auth/src/vmq_enhanced_auth.erl

```diff
@@ -346,7 +362,10 @@ match(TIn, T, Tbl, Type, Key, Qos) ->
             case ets:lookup(Tbl, Key) of
                 [{_, _, Label}] ->
                     check_label_and_incr_metrics(Label, Type, Qos),
-                    true;
+                    MatchedAcl = #matched_acl{
+                        name = Label, pattern = iolist_to_binary(vmq_topic:unword(T))
+                    },
+                    {true, MatchedAcl};
                 _ ->
                     true
```

Comment:
Given we are handling only the cases where match fn returns {true, MatchedACL} or false, can we handle this case as well? Here it should not return true but {true, #matched_acl{}} IMO. CMIIW.